src/OrbitBase/include/OrbitBase/SharedState.h

```diff
@@ -16,9 +16,10 @@
 
 namespace orbit_base_internal {
 
-// SharedState<T> is an implementation detail of the Future<T> / Promise<T> facility.
+// SharedState<T> is an implementation detail of the Future<T> / Promise<T> / StopSource / StopToken
+// facility.
 //
-// Don't use this class outside of Promise<T> / Future<T>!
+// Don't use this class outside of Promise<T> / Future<T> / StopSource / StopToken!
```

Comment:
I think you shouldn't reuse this SharedState class but rather define your own. Feel free to rename this one to make it more clear it's for promises and futures.

Reply:
done. Called my class StopState. Did not rename SharedState